core/src/main/java/org/jobrunr/server/tasks/startup/CheckIfAllJobsExistTask.java

```diff
@@ -47,10 +51,21 @@ public void run() {
     }
 
     private Set<String> getDistinctRecurringJobSignaturesThatDoNotExistAnymore() {
-        return storageProvider.getRecurringJobs().stream()
-                .map(AbstractJob::getJobSignature)
-                .filter(jobSignature -> !jobExists(jobSignature))
-                .collect(toSet());
+        Set<String> missingRecurringJobSignatures = new HashSet<>();
+        for (RecurringJob recurringJob : storageProvider.getRecurringJobs()) {
+            if (ANNOTATION.equals(recurringJob.getCreatedBy())) {
+                if (!jobExists(recurringJob.getJobSignature())) {
+                    storageProvider.deleteRecurringJob(recurringJob.getId());
+                    LOGGER.info("Deleting recurring job {} ({}) as it was created by the @Recurring annotation but does not exist anymore", recurringJob.getId(), recurringJob.getJobSignature());
+                } else if (isNotPresent(getRecurringAnnotation(recurringJob.getJobDetails()))) {
+                    storageProvider.deleteRecurringJob(recurringJob.getId());
+                    LOGGER.info("Deleting recurring job {} ({}) as it was created by the @Recurring annotation but is not annotated by the  @Recurring annotation anymore", recurringJob.getId(), recurringJob.getJobSignature());
```

Comment:
Should the log be above `storageProvider.deleteRecurringJob(recurringJob.getId());`? Or changed to

```suggestion
                    LOGGER.info("Deleted recurring job {} ({}) as it was created by the @Recurring annotation but is not annotated by the  @Recurring annotation anymore", recurringJob.getId(), recurringJob.getJobSignature());
```